server/src/main/java/org/apache/druid/server/coordinator/helper/DruidCoordinatorBalancer.java

```diff
@@ -220,6 +245,7 @@ protected void moveSegment(
             segmentToMove,
```

Comment:
Interesting that `DruidCoordinator.moveSegment()` also wraps pretty much it's whole body with `try { ... } catch (Exception e) { ...}`, that will makes `catch (Exception e)` below in *this* method obsolete. I think only one should be left.